plotly/tools.py

```diff
@@ -51,6 +50,15 @@ def warning_on_one_line(message, category, filename, lineno,
 except ImportError:
     _numpy_imported = False
 
+try:
+    import scipy as scp
+    import scipy.spatial as scs 
+    import scipy.cluster.hierarchy as sch 
+
+    _scipy_imported = True
+except ImportError:
+    _scipy_imported = False
```

Comment:
I prefer protected imports as a user  for sure. 
refined in 44cdece2c3b91c8cb57ad0ddb3fff9283ff0f989